### ISSUE TITLE
The Jira Marketplace button located at Self Hosted > Settings > Integrations > Jira directs to the Jira app for cloud hosted sentry.

### ISSUE THREAD

**Issue by edgariscoding** [getsentry/sentry#61702]
### Self-Hosted Version

23.11.0

### CPU Architecture

x86

### Docker Version

20.10

### Docker Compose Version

2.6

### Steps to Reproduce

Go to Self Hosted > Settings > Integrations > Jira and click on the Jira Marketplace button at the top right.

### Expected Result

Expected to be able to set up Jira integration for my self hosted Sentry.

### Actual Result

I installed the Jira app and at the end I was taken to sentry.io to log in, which is obviously incorrect.

**Comment by Dhrumil-Sentry** [comment]
@hubertdeng123 can you please help here

**Comment by hubertdeng123** [comment]
I am not familiar with this area of code, so this item will need to be backlogged if it indeed is broken as is.

**Comment by franzos** [comment]
I just dealt with this issue myself.

1. Check everything is setup properly `https://my-sentry-install.com/extensions/jira/descriptor/`
2. Add URL to Jira (Manage Apps > Add application) like [here](https://forum.sentry.io/t/how-to-configure-jira-cloud-in-your-on-premise-sentry/6720)
3. Jira will do some post-install stuff, which ultimately fails (I'm now at `https://my-project-name.atlassian.net/plugins/servlet/ac/my-sentry-install.com.jira/post-install-sentry` in my browser)

Looking at the developer the browser logs, I found something like this:

```
The loading of “https://my-sentry-install.com/extensions/jira/ui-hook/?xdm_e=https%3A%2F%2Fmy-project.atlassian.net&xdm_c=my-sentry-install.com.jira__post-install-sentry&cp=&xdm_deprecated_addon_key_do_not_use=sentry.pantherx.dev.jira&lic=none&cv=1001.0.0-SNAPSHOT&jwt=eyJ0eXAiOiJKV1QiLCJhbG....” in a frame is denied by “X-Frame-Options“ directive set to “deny“.
```

Looked something like this:

![swappy-20240416_083908](https://github.com/getsentry/sentry/assets/2619225/28f23e92-b07f-472d-adc7-cb0bfe72dee8)

Clicking the link related to the denied directive, brought me back to my self-hosted sentry instance, where the integration was now complete.

**Comment by danstreeter** [comment]
I'm having the same issue on `24.6.0.dev0` that I installed today.

Neither installing the Sentry Marketplace app within Jira (which redirects to the cloud hosted Sentry.io instance)
or the method that @franzos mentioned above (as my Jira instance has no 'Upload App' anymore as per the docs referenced work.

:(

**Comment by GabeVillalobos** [comment]
Unfortunately the team is understaffed and we can’t prioritize this fix, but we’re always open to community contributions.

**Comment by danstreeter** [comment]
> ...(as my Jira instance has no 'Upload App' anymore as per the docs referenced work.)

Just got the closed state for this issue and reminded me to check things - Since my update above, the "Upload App" option on my Jira instance had re-appeared and enabled me to add it that way and my integration was added and works. Apologies I didnt come back and update here sooner. Not sure when I did it - or when Atlassian upgraded my Jira cloud version to support the "Upload App" again.
